core/zip/src/ZInflate.c

```diff
@@ -1183,7 +1199,7 @@ void R__unzip(int *srcsize, uch *src, int *tgtsize, uch *tgt, int *irep)
   /*   D E C O M P R E S S   D A T A  */
 
   /* New zlib format */
-  if (src[0] == 'Z' && src[1] == 'L') {
+  if (is_valid_header_zlib(src)) {
```

Comment:
This is as the side effect of adding one comparison ... which actually lead me to ponder what the semantic of the (src[2] == Z_DEFLATED) test is ... and why/whether there isn't an equivalent for LZ4.